### DISCUSSION TITLE
Periods of activity

### DISCUSSION THREAD

**Discussion by rpiening** [schemaorg/schemaorg#3370] [Q&A]
Hi all,

I'm rather new to schema.org and wondering if there is a standard way to decribe a Person's period of activity. 

My data happens to be specifically about artists, I want to model both their general period of activity (e.g. Rembrandt made artworks between 1625 and 1669) and the specific periods they were working in one place (e.g. Rembrandt worked in Leiden between 1625 and 1633). 

This seems like a question that would also come up in the context of personal bios/curriculum vitae, e.g. "person worked at company X from 2005 to 2007". 

I thought the following might express what I want:
```
example/rembrandt
 a sdo:Person ;
 sdo:name "Rembrandt" ;
 sdo:performerIn [
  a sdo:Event ;
  sdo:additionalType <http://vocab.getty.edu/aat/300393177> ; # active
  sdo:startDate "1625" ;
  sdo:endDate "1669"
 ] .
```
 
  Is this the way to go or am I overlooking an already set standard or property here?
  Thanks in advance!

**Comment by RichardWallis** [comment]
Check out the [Role](https://schema.org/Role) type and the following blog posts: [Introducing 'Role'](http://blog.schema.org/2014/06/introducing-role.html) & [The role of Role in Schema.org](https://www.dataliberate.com/2015/04/15/the-role-of-role-in-schema-org/).

**Reply by rpiening** [reply]
Thank you Richard! Those are some very informative blog posts. The Role type would give me a neat way to model the dates of activity. 

For reference, this is what I have arrived at:
```
<example.org/rembrandt>
    a sdo:Person ;
    sdo:name "Rembrandt" ;
    sdo:hasOccupation [
        a sdo:Role ;
        sdo:roleName <http://vocab.getty.edu/aat/300393177> ; # active
        sdo:startDate "1625" ;
        sdo:endDate "1669" ;
        sdo:hasOccupation [
            a sdo:Occupation ;
            sdo:name "Period of activity"
        ] ;
```

Do you perhaps also have a suggestion for how to express the location of the Role (or Occupation, as that seems to be the most appropriate type here)? Neither type has an appropriate 'location' property. 

Alternatively, I'm looking at using workLocation to express separate 'locations of activity', but then again, I need some way to put dates onto that. Could Role be used for that as well? I.e.:
```
<example.org/rembrandt>
    a sdo:Person ;
    sdo:name "Rembrandt" ;
    sdo:workLocation [
        a sdo:Role ;
        sdo:startDate "1625" ;
        sdo:endDate "1633" ;
        sdo:workLocation <https://sws.geonames.org/6950266> # Leiden
    ] .
```
Many thanks!

**Reply by RichardWallis** [reply]
@rpiening As [Occupation](https://schema.org/Occupation) has a [occupationLocation](https://schema.org/occupationLocation) property you could flesh out the description of the _Occupation_ entity in your first example.  Your second example makes sense.

One word of caution. It is up to the consumers of your data as to if they recognise and use the complexity of data you provide, even though it is produced in line with Schema.org documentation and guidance.  Having said that, the more varied Schema.org patterns are found in the wild, the more likely data consumers are to recognise them.  Personally, if I have the data and it is easy to publish it, I publish it.
